pinot-core/src/main/java/org/apache/pinot/core/segment/index/loader/defaultcolumn/BaseDefaultColumnHandler.java

```diff
@@ -285,6 +291,89 @@ protected void removeColumnV1Indices(String column)
     SegmentColumnarIndexCreator.removeColumnMetadataInfo(_segmentProperties, column);
   }
 
+  /**
+   * Right now the text index is supported on RAW (non-dictionary encoded)
+   * single-value STRING columns. Eventually we will relax the constraints
+   * step by step.
+   * For example, later on user should be able to create text index on
+   * a dictionary encoded STRING column that also has native Pinot's inverted
+   * index. We can also support it on BYTE columns later.
+   * @param column column name
+   * @param indexLoadingConfig index loading config
+   * @param fieldSpec field spec
+   */
+  private void checkUnsupForEnablingTextIndexOnNewColumn(String column, IndexLoadingConfig indexLoadingConfig,
+      FieldSpec fieldSpec) {
+    if (!indexLoadingConfig.getNoDictionaryColumns().contains(column)) {
+      throw new UnsupportedOperationException("Text index is currently not supported on dictionary encoded columns");
```

Comment:
You should include the column name in each of these exceptions.
Alternatrively, it may be better to return a boolean, and use PreConditions in the caller and include the column name in the exception thrown.

Reply:
Included the column name